homeassistant/components/switch/recswitch.py

```diff
@@ -0,0 +1,89 @@
+"""
+Support for Ankuoo RecSwitch MS6126 devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.recswitch/
+"""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA, SwitchDevice
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.recswitch import DATA_RSH
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['recswitch']
+
+DEFAULT_NAME = 'RecSwitch {0}'
+
+POLLING_INTERVAL = timedelta(seconds=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_MAC): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Setup the device and register on recswitch network."""
+    ip_address = config.get(CONF_HOST)
+    mac_address = config.get(CONF_MAC).upper()
+    device_name = config.get(CONF_NAME)
+
+    device = hass.data[DATA_RSH].register_device(mac_address, ip_address)
+    async_add_entities([RecSwitchSwitch(device, device_name, mac_address)])
+
+
+class RecSwitchSwitch(SwitchDevice):
+    """Representation of a recswitch device."""
+
+    def __init__(self, device, device_name, mac_address):
+        self.gpio_state = self.assumed_state
+        self.device = device
+        self.device_name = device_name
+        self.device.report_gpio_change = self.report_gpio_change
```

Comment:
If there's a callback upon state change from the device, shouldn't this be a non polling entity?

Reply:
Move the registering of the callback to `async_added_to_hass` entity coroutine.

Reply:
Sometime the device seems to report the status change.. but really few times..